src/Messaging/src/Messaging/MessageHeaders.cs

```diff
@@ -357,4 +358,29 @@ protected void UpdateHeaders(string id, long? timestamp)
             Headers[TimestampName] = timestamp.Value;
         }
     }
+
```

Comment:
This seems to go to extraordinary lengths - The justification for the rule is that somehow at runtime supplying an invalid type could cause a runtime error (correct me if I am wrong) but in this case, we are going from known interface to known interface and can validate that it works .